conf/nomad/presto.hcl

```diff
@@ -310,7 +313,9 @@ http-server.https.truststore.path=/local/roots.pem
 
 # This is the same jks, but it will not do the consul connect authorization in intra cluster communication
 internal-communication.https.required=true
-internal-communication.shared-secret=${cluster_shared_secret}
+{{ with secret "secret/data/presto" }}
+internal-communication.shared-secret="{{ .Data.data.cluster_shared_secret}}"
+{{end}}
```

Comment:
Could it be fetched via terraform-provider-vault and injected via the input variable? I am not sure that coupling with the secret path is a good idea

Reply:
I implemented it this way as the description of the issue reads as :
"This secret should be added to vault and used by presto by referencing vault in a template stanza."
https://www.nomadproject.io/docs/job-specification/template#vault-kv-api-v2

Reply:
@Neha-Sinha2305 I agree :)